presto-main/src/main/java/com/facebook/presto/sql/analyzer/ScopeFactory.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.analyzer;
+
+import com.facebook.presto.sql.tree.Node;
+
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class ScopeFactory
```

Comment:
I removed it. I had it to automatically add `Scope` to `Analysis` when it is created. But it was not nice. I didn't like it either. I created a `Scope.Builder` instead and added `createScope` methods to `StatementAnalyzer`. Which is much better, although the builder usage is quite verbose.